src/test/java/com/commercetools/sync/products/helpers/variantreferenceresolver/WithCategoryReferencesTest.java

```diff
@@ -0,0 +1,263 @@
+package com.commercetools.sync.products.helpers.variantreferenceresolver;
+
+import com.commercetools.sync.products.ProductSyncOptions;
+import com.commercetools.sync.products.ProductSyncOptionsBuilder;
+import com.commercetools.sync.products.helpers.VariantReferenceResolver;
+import com.commercetools.sync.services.CategoryService;
+import com.commercetools.sync.services.ChannelService;
+import com.commercetools.sync.services.CustomerGroupService;
+import com.commercetools.sync.services.ProductService;
+import com.commercetools.sync.services.TypeService;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.sphere.sdk.categories.Category;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.products.ProductVariantDraft;
+import io.sphere.sdk.products.ProductVariantDraftBuilder;
+import io.sphere.sdk.products.attributes.AttributeDraft;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.Spliterator;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import static com.commercetools.sync.products.ProductSyncMockUtils.createReferenceObject;
+import static com.commercetools.sync.products.ProductSyncMockUtils.getMockCategoryService;
+import static com.commercetools.sync.products.ProductSyncMockUtils.getReferenceSetAttributeDraft;
+import static com.commercetools.sync.products.helpers.VariantReferenceResolver.REFERENCE_ID_FIELD;
+import static com.commercetools.sync.products.helpers.VariantReferenceResolver.REFERENCE_TYPE_ID_FIELD;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class WithCategoryReferencesTest {
+    private CategoryService categoryService;
+    private static final String CATEGORY_ID = UUID.randomUUID().toString();
+    private VariantReferenceResolver referenceResolver;
+
+    /**
+     * Sets up the services and the options needed for reference resolution.
+     */
+    @BeforeEach
+    void setup() {
+        categoryService = getMockCategoryService(CATEGORY_ID);
+        final ProductSyncOptions syncOptions = ProductSyncOptionsBuilder.of(mock(SphereClient.class)).build();
+        referenceResolver = new VariantReferenceResolver(syncOptions,
+            mock(TypeService.class),
+            mock(ChannelService.class),
+            mock(CustomerGroupService.class),
+            mock(ProductService.class),
+            categoryService);
+    }
+
+    @Test
+    void resolveReferences_WithNonExistingCategoryReferenceAttribute_ShouldReturnEqualDraft() {
```

Comment:
to make it consistent use ShouldNotResolveReferences instead of ShouldReturnEqualDraft.

Reply:
if you agree to change it, could you please check this method name in all PR?

Reply:
yes, I agree.

Reply:
aa8c3f8